pkg/broker/reconstruct.go

```diff
@@ -0,0 +1,83 @@
+package broker
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/internal/constants"
+	"github.com/submariner-io/subctl/internal/rbac"
+	"github.com/submariner-io/subctl/pkg/brokercr"
+	"github.com/submariner-io/subctl/pkg/cluster"
+	"github.com/submariner-io/submariner-operator/api/v1alpha1"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	controllerClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ReconstructData(
+	brokerClusterInfo, submClusterInfo *cluster.Info, namespace string, status reporter.Interface,
+	ipsecSubmFile string) error {
+	status.Start("Retrieving data to reconstruct broker-info.subm")
+    defer status.End()
+
+	data := &Info{}
+
+	kubeClient, err := kubernetes.NewForConfig(brokerClusterInfo.RestConfig)
+	if err != nil {
+		return status.Error(err, "error creating Kubernetes client")
+	}
+
+	data.BrokerURL = brokerClusterInfo.RestConfig.Host + brokerClusterInfo.RestConfig.APIPath
+
+	data.ClientToken, err = rbac.GetClientTokenSecret(context.TODO(), kubeClient, namespace, constants.SubmarinerBrokerAdminSA)
+	if err != nil {
+		return status.Error(err, "error getting broker client secret")
+	}
+
+	broker := &v1alpha1.Broker{}
+	err = brokerClusterInfo.ClientProducer.ForGeneral().Get(context.TODO(), controllerClient.ObjectKey{
+		Namespace: namespace,
+		Name: brokercr.Name,
+	}, broker)
+
+	if err == nil {
+		data.Components = broker.Spec.Components
+		data.ServiceDiscovery = data.IsServiceDiscoveryEnabled()
+		data.CustomDomains = &broker.Spec.DefaultCustomDomains
+	} else if !apierrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+		return status.Error(err, "error retrieving Broker")
+	}
+
+	if submClusterInfo.Submariner != nil {
+		status.Success("Retrieving IPSec PSK secret from Submariner found on cluster %s", submClusterInfo.Name)
+		data.IPSecPSK = &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ipsecPSKSecretName,
+			},
+			Data: map[string][]byte{"psk": []byte(submClusterInfo.Submariner.Spec.CeIPSecPSK)},
+		}
+	} else if ipsecSubmFile != "" {
+		status.Success("Retrieving IPSec PSK from the file %s", ipsecSubmFile)
+		ipsecData, err := ReadInfoFromFile(ipsecSubmFile)
+		if err != nil {
+			return errors.Wrapf(err, "error importing IPsec PSK secret from file %q", ipsecSubmFile)
+		}
+
+		data.IPSecPSK = ipsecData.IPSecPSK
+	} else {
+		status.Success("Generating new IPSec PSK secret")
```

Comment:
This should be a warning, because the new file won’t be usable to connect existing clusters.